pandas/io/sql.py

```diff
@@ -1040,10 +1040,7 @@ def insert_data(self) -> tuple[list[str], list[np.ndarray]]:
                         # GH#53854 to_pydatetime not supported for pyarrow date dtypes
                         d = ser._values.to_numpy(dtype=object)
                     else:
-                        with warnings.catch_warnings():
-                            warnings.filterwarnings("ignore", category=FutureWarning)
-                            # GH#52459 to_pydatetime will return Index[object]
-                            d = np.asarray(ser.dt.to_pydatetime(), dtype=object)
+                        d = ser.dt.to_pydatetime().to_numpy(dtype=object)
```

Comment:
@phofl what's the way to make this writable later on (`d[mask] = None` below)?

Reply:
Is 
```
ser.dt.to_pydatetime()
```

ever not backed by a NumPy array? If it's always a NumPy array I would just do

```
ser.dt.to_pydatetime()._values
```

Otherwise you can do

d.flags.writeable = True, but this is a bit tricky, because this won't work if d was initially a PyArrow array

Reply:
In this branch, `ser` was backed by `ArrowExtensionArray` so it's initially a pyarrow array

Reply:
But the result of to_pydatetime is always backed by numpy, correct?

Reply:
Ah OK yes it is